app/src/Controller/DomainController.php

```diff
@@ -260,8 +260,12 @@ public function edit(Request $request, Domain $domain, FileUploader $fileUploade
             return $this->redirectToRoute('domain_index');
         }
 
-        $dkim = $this->em->getRepository(DomainKey::class)->find($domain->getDomainKeys()->getId());
-        $dnsInfo = $dkim->getDnsinfo();
+        $dkim = null;
+        $dnsInfo = null;
+        if ($domain->getDomainKeys() !== null) {
+            $dkim = $this->em->getRepository(DomainKey::class)->find($domain->getDomainKeys()->getId());
+            $dnsInfo = $dkim->getDnsinfo();
+        }
```

Comment:
`$domain->getDomainKeys()` te retourne déjà l’objet `$dkim`, donc il suffit de faire :

```suggestion
        $dkim = $domain->getDomainKeys();
        $dnsInfo = $dkim?->getDnsinfo();
```

(note le `$dkim?->getDnsinfo()` qui retourne `null` si `$dkim` est `null`)